demos/common/mqtt_demo_helpers/mqtt_demo_helpers.c

```diff
@@ -604,9 +600,9 @@ BaseType_t SubscribeToTopic( MQTTContext_t * pxMqttContext,
                 xBackoffStatus = prvBackoffForRetry( &xRetryParams );
             }
         }
-    } while( ( xFailedSubscribeToTopic == pdTRUE ) && ( xBackoffStatus == pdPASS ) );
+    } while( ( xSubscribeStatus == pdFAIL ) && ( xBackoffStatus == pdPASS ) );
```

Comment:
Wouldn't this result in retrying subscription request when the response is never received from the server? Should we just break out of the loop if a response is not received from the broker?

Reply:
Yes, it will fail and end up retrying. It should just break out of the loop when `xSubscribeStatus == pdFAIL`.